cli/command/task_test.go

```diff
@@ -0,0 +1,295 @@
+package command
+
+import (
+	"testing"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+)
+
+func TestCreateTask(t *testing.T) {
+	testWaitHelper(t, func(t *testing.T, wait bool) {
+		base, ctrl := newTestCommand(t)
+		defer ctrl.Finish()
+		taskCommand := NewTaskCommand(base.Command())
+		args := Args{"env_name", "dpl_name", "task_name"}
+		flags := Flags{
+			"copies": 1,
+			"env":    []string{"container:key=val"},
+		}
+		overrides := []models.ContainerOverride{{
+			ContainerName:        "container",
+			EnvironmentOverrides: map[string]string{"key": "val"},
+		}}
+
+		req := models.CreateTaskRequest{
+			TaskName:           "task_name",
+			DeployID:           "dpl_id",
+			EnvironmentID:      "env_id",
+			ContainerOverrides: overrides,
+		}
+
+		base.Resolver.EXPECT().
+			Resolve("deploy", "dpl_name").
+			Return(Args{"dpl_id"}, nil)
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name").
+			Return(Args{"env_id"}, nil)
+
+		base.Client.EXPECT().
+			CreateTask(req).
+			Return("job_id", nil)
+
+		if wait {
+			base.Client.EXPECT().
+				ReadJob("job_id").
+				Return(&models.Job{
```

Comment:
we typically don't add multi-line objects in our mock-chains since it makes things so hard to read. 
Also, don't use raw strings for the `job.Status` field (or `job.JobType` if you ever encounter that) - we have special types in the `api/job` package for that. 

For example, this would be fine: 
```
base.Client.EXPECT().
    ReadJob("job_id").
    Return(&models.Job{JobD: "job_id"}, nil)
```

but this should be:
```
job := &models.Job{
    Status: job.Completed.String(),
    Result: "task_id",
}

base.Client.EXPECT().
    ReadJob("job_id").
    Return(job, nil)
```